README.md

```diff
@@ -0,0 +1,217 @@
+|------------------------------|------------------------------------------|
+| `REVERT_TESTING`             | The initial stage when running a revert. |
+| `REVERT_TESTING_FAILED`      | `REVERT_TESTING` failed.                 |
+| `REVERT_MIGRATING`           | Revert is being executed.                |
+| `REVERT_MIGRATION_FAILED`    | `REVERT_MIGRATING` failed.               |
+| `REVERT_VERIFYING`           | Verifying that a revert succeeded.       |
+| `REVERT_VERIFICATION_FAILED` | The revert failed.                       |
+| `REVERT_SUCCEEDED`           | The revert succeeded.                    |
+
+Reverting a migration is useful for quick rollback of a failed. Also, each
+state is also accompanied by a free-text `payload` containing additional
+information.
+
+The following state transitions exist:
+
+| Name | Source states | Destination state |
+|------|---------------|-------------------|
+| migrate | `NEW` | `TESTING` |
+| fail-testing | `TESTING` | `TESTING_FAILED` |
+| execute-migration | `TESTING` | `MIGRATING` |
+| fail-migration | `MIGRATING` | `MIGRATION_FAILED` |
+| verify-migration | `MIGRATING` | `VERIFYING`
+| fail-verification | `VERIFYING` | `VERIFICATION_FAILED` |
+| succeed-migration | `VERIFYING` | `MIGRATION_SUCCEEDED` |
+| retry-migration | `TESTING_FAILED`, `MIGRATION_FAILED`, `VERIFICATION_FAILED` | `TESTING` |
+| revert-migration | `MIGRATION_SUCCEEDED` | `REVERT_TESTING` |
+| fail-revert-testing | `REVERT_TESTING` | `REVERT_TESTING_FAILED` |
+| execute-revert | `REVERT_TESTING` | `REVERT_MIGRATING` |
+| fail-revert-execution | `REVERT_MIGRATING` | `REVERT_MIGRATION_FAILED` |
+| verify-revert | `REVERT_MIGRATING` | `REVERT_VERIFYING` |
+| fail-revert-verification | `REVERT_VERIFYING` | `REVERT_VERIFICATION_FAILED` |
+| succeed-migraton | `REVERT_VERIFYING` | `REVERT_SUCCEEDED` |
+| retry-revert | `REVERT_TESTING_FAILED`, `REVERT_MIGRATION_FAILED`, `REVERT_VERIFICATION_FAILED` | `REVERT_TESTING` |
+
+Getting started
+---------------
+
+First, [install Go](https://golang.org/dl/) and then set the `GOPATH`
+environment variable to a directory where you'd like to install the binaries.
+You can then install the binaries needed for this example by issuing
+
+```bash
+$ go install github.com/any-migrate/any-migrate
+$ go install github.com/any-migrate/file-repository
+$ go install github.com/any-migrate/mysql-migrator
+```
+
+`any-migrate` is the command line utility that you'll use to interact with
+any-migrate. A "repository" is a plugin which is used for keeping track of
+which migrations have been applied. The "file" stores this informtion in a
+plain file. A migrator is a plugin which applies changes. The "mysql" plugin
+applies changes to a MySQL database.
+
+Then create a configuration (YAML file):
+
+```yaml
+# my.config
+repository:
+  type: file
+  path: ./my.repository
+migrations:
+  path: ./my.changes
+sources:
+  mysql:
+    migrator: mysql-migrator
```

Comment:
One thing that I really hate is when something works on development/staging but doesn't work on production. I think that it is important that this software tries to minimze this as much as possible.  

Two things that I think about that often differs between environments.

- Versions of software (for example mysql versions)
- Access (for example granted rights in mysql)

Access can be a bit tricky but I could see a point in having a 1A and 1B version of a migration where A version is applied to versions running version `A` and B is applied to versions running `B`. 

Specifying a migration on the form will make it easier to check access compared to a hard coded sql script. 
```
type: rename table
from: ussers
to: users
```